src/frontend/src/optimizer/plan_node/stream_topn.rs

```diff
@@ -49,13 +49,21 @@ impl StreamTopN {
 
 impl fmt::Display for StreamTopN {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        write!(
-            f,
-            "StreamTopN {{ order: {}, limit: {}, offset: {} }}",
-            self.logical.topn_order(),
-            self.logical.limit(),
-            self.logical.offset(),
-        )
+        let mut builder = if self.input().append_only() {
+            f.debug_struct("StreamAppendOnlyTopN")
+        } else {
+            f.debug_struct("StreamTopN")
+        };
```

Comment:
It looks duplicated with "append_only = true/false" in L62~64

(Similar for other operators)

Reply:
The `append_only ` attribute in `PlanBase` indicates whether the output of operator is append-only, it doesn't mean the operator is append-only optimized. So if we want to check whether the append-only version operator is used, we should differentiate it by name (likes the `StreamDeltaJoin` and `StreamHashJoin`).

Reply:
Got it. In this way, I would recommend to remove  "append_only = true/false" in explain results.

Reply:
agreed. I got it wrong before.